sql/src/test/java/io/crate/integrationtests/CrateMetaDataUpgradeServiceTest.java

```diff
@@ -73,41 +78,76 @@ public void shutdown() throws IOException {
+        TestingHelpers.assertCrateVersion(response.rows()[1][1], null, Version.CURRENT);
+
+        execute("select routing_hash_function, version " +
+                "from information_schema.table_partitions "+
+                "where table_name = 'testneedsupgrade_parted'");
+        assertThat(response.rowCount(), is(5L));
+        for (Object[] row : response.rows()) {
+            assertThat(row[0], is("org.elasticsearch.cluster.routing.DjbHashFunction"));
+            TestingHelpers.assertCrateVersion(row[1], null, Version.CURRENT);
+        }
     }
 
     @Test
-    public void testNewRoutingHashFunction() throws Exception {
+    public void testAlreadyUpgradedTables() throws Exception {
         startUpNodeWithDataDir("/indices/data_home/cratedata_already_upgraded.zip");
-        execute("select routing_hash_function, table_name from information_schema.tables "+
-                "where table_name in ('testalreadyupgraded', 'testalreadyupgraded_parted') order by 2");
-        assertThat(TestingHelpers.printedTable(response.rows()),
-            is ("org.elasticsearch.cluster.routing.Murmur3HashFunction| testalreadyupgraded\n" +
-                "org.elasticsearch.cluster.routing.Murmur3HashFunction| testalreadyupgraded_parted\n"));
-
-        execute("select routing_hash_function, partition_ident from information_schema.table_partitions "+
-                "where table_name = 'testalreadyupgraded_parted' order by 2");
-        assertThat(TestingHelpers.printedTable(response.rows()),
-            is ("org.elasticsearch.cluster.routing.Murmur3HashFunction| 04132\n" +
-                "org.elasticsearch.cluster.routing.Murmur3HashFunction| 04134\n" +
-                "org.elasticsearch.cluster.routing.Murmur3HashFunction| 04136\n" +
-                "org.elasticsearch.cluster.routing.Murmur3HashFunction| 04138\n" +
-                "org.elasticsearch.cluster.routing.Murmur3HashFunction| 0413a\n"));
+        execute("select routing_hash_function, version " +
+                "from information_schema.tables "+
+                "where table_name in ('testalreadyupgraded', 'testalreadyupgraded_parted') order by table_name");
+        assertThat(response.rowCount(), is(2L));
+        for (Object[] row : response.rows()) {
+            assertThat(row[0], is("org.elasticsearch.cluster.routing.Murmur3HashFunction"));
+            TestingHelpers.assertCrateVersion(row[1], null, Version.CURRENT);
+        }
+
+        execute("select routing_hash_function, version " +
+                "from information_schema.table_partitions "+
+                "where table_name = 'testalreadyupgraded_parted'");
+        assertThat(response.rowCount(), is(5L));
+        for (Object[] row : response.rows()) {
+            assertThat(row[0], is("org.elasticsearch.cluster.routing.Murmur3HashFunction"));
+            TestingHelpers.assertCrateVersion(row[1], null, Version.CURRENT);
+        }
+    }
+
+    @Test
+    public void testNewTables() throws Exception {
```

Comment:
remove, this has nothing to do with the upgrade service. its already tested by an `CreateAlterTableStatementAnalyzerTest` test